lib/stdlib/src/shell_docs_test.erl

```diff
@@ -154,9 +154,12 @@ test([H | T], Bindings) ->
 test(Text, _Bindings) when is_binary(Text); Text =:= [] ->
     [].
 
+-define(RE_CAPTURE, "(?:(?'line_number'[0-9]+(?'prefix'>\s))|(?'prefix'%))?(?'content'.*)").
+-define(RE_OPTIONS, [ {capture, [line_number, prefix, content] ,binary}, dupnames ]).
 run_test(Code, InitialBindings) ->
     Lines = string:split(Code, "\n", all),
-    Tests = inspect(parse_tests(Lines, [])),
+    ReLines = [ re:run(Line, ?RE_CAPTURE, ?RE_OPTIONS) || Line <- Lines],
+    Tests = inspect(parse_tests(ReLines, [])),
     lists:foldl(fun(Test, Bindings) ->
                         run_tests(Test, Bindings)
                 end, InitialBindings, Tests).
```

Comment:
The first line needs to be a match, otherwise the code should not be tested. The tests in lists_SUITE + binary_module_SUITE are failing as code examples that are not meant to be tested are. Besides that I think the changes look good.

Would you mind adding some extra nagative-tests to this module? For example:

````
```
a> should not be tested
```

```
 1> should not be tested
```

```
> should not be tested
```

```
should not be tested
1> 
```

```
% should probably be tested?
1> ok.
```

```
% should probably be tested?

1> ok.
```

````

Reply:
I've added the extra tests, which if I disable the `Prebound` variable test seems to work fine with recent code changes, eg.

````diff
 ## Prebound variables:
 
 ```
-1> Prebound.
+> Prebound.
 hello
 ```
````

...  I'm none too happy with adding a `case` statement where I did, but it seems to provide similar behavior of matching first line was what was there prior to my attentions;.

```diff
-test({pre,[],[{code,Attrs,[<<">",_/binary>> = Code]}]}, Bindings) ->
+test({pre,[],[{code,Attrs,[Code]}]}, Bindings) when is_binary(Code) ->
```